src/environment/stella_environment.hpp

```diff
@@ -64,6 +66,18 @@ class StellaEnvironment {
       */
     reward_t act(Action player_a_action, Action player_b_action);
 
+    /** This functions emulates a push on the reset button of the console */
+    void softReset();
+
+    /** Keep pressing the console select button for a given amount of time*/
+    void pressSelect(size_t num_steps = 1);
+
+    /** Set the difficulty according to the mask.
+      * If the first bit is 1, then it will put the left difficulty switch to A (otherwise leave it on B)
+      * If the second bit is 1, then it will put the right difficulty switch to A (otherwise leave it on B)
+      */
+    void setDifficulty(difficulty_t mask);
```

Comment:
See comments above.

Reply:
I moved `m_difficulty` to `ale_state`, but I still do a couple of things inside the `StellaEnvironment` after I set the difficulty, namely:

`m_osystem->console().mediaSource().update();`
`processScreen();`
`processRAM();`

Do you want me to get rid of this whole function?

Reply:
So setDifficulty() really is "toggle the difficulty switch and take a step in the environment", right? No, that seems like the right place for the remaining code.